### DISCUSSION TITLE
Why do examples and docs not show using context managers for the producer and consumer?

### DISCUSSION THREAD

**Discussion by MicaelJarniac** [aio-libs/aiokafka#925] [Q&A] [answered]
On the docs and examples, I've only seen it done like this:
```py
producer = aiokafka.AIOKafkaProducer(bootstrap_servers="localhost:9092")
await producer.start()
try:
    await producer.send_and_wait("my_topic", b"Super message")
finally:
    await producer.stop()
```
But from reading the code and testing, the following also seems to be possible:
```py
async with aiokafka.AIOKafkaProducer(bootstrap_servers="localhost:9092") as producer:
    await producer.send_and_wait("my_topic", b"Super message")
```
The same goes for the consumer.

I wanted to know why is this omitted from the docs and examples. Is there a drawback? Is it not recommended?

I personally would prefer using context managers, as it's fewer lines to write, and seems more idiomatic.

**Comment by ods** [comment, marked as answer]
The `async with` approach is usually preferable. It was just added later and documentation hasn't been updated to reflect it.
